### PR TITLE
Fixed SecretDetector Bug for None String

### PR DESCRIPTION
**Pull request by sfc-gh-kwagner**
Fixed SecretDetector Bug for None String. Also got secret detector to 100% code cov

### CODE REVIEW

**Review by sfc-gh-mkeller — APPROVED** [review]
Great work @sfc-gh-kwagner 🚢 🚢 
Make sure that @sfc-gh-wshangguan reviews before merging though

File: src/snowflake/connector/secret_detector.py
Comment (sfc-gh-stakeda) [inline src/snowflake/connector/secret_detector.py:74]: `telemetry.py` looks like expect this returns a str.

Comment (sfc-gh-wshangguan) [reply src/snowflake/connector/secret_detector.py:74]: +1

Comment (sfc-gh-kwagner) [reply src/snowflake/connector/secret_detector.py:74]: Thats a bug then. The secret detector has always returned a tuple. I'll fix that in telemetry.py and look for other misuses

Comment (sfc-gh-wshangguan) [reply src/snowflake/connector/secret_detector.py:74]: That might be a wrong usage of telemetry.py. Could you please help to fix that in telemetry.py? @sfc-gh-kwagner

Comment (sfc-gh-kwagner) [reply src/snowflake/connector/secret_detector.py:74]: @sfc-gh-stakeda @sfc-gh-wshangguan Fixed

**Comment by sfc-gh-wshangguan** [comment]
Would be better to get review from author as well. Invited @sfc-gh-hchaturvedi .

**Review by sfc-gh-stakeda — APPROVED** [review]

**Review by sfc-gh-wshangguan — APPROVED** [review]
LGTM, that's a nice catch!

**Comment by sfc-gh-stakeda** [comment]
sounds like the python's type checking doesn't work for all cases...

**Comment by sfc-gh-kwagner** [comment]
> sounds like the python's type checking doesn't work for all cases...

@sfc-gh-stakeda from what I understand, Python's type checking is really just a docstring. Doesn't cause any compilation or build issues but is more for the IDE and new developers using that method

**Comment by sfc-gh-wshangguan** [comment]
Agree with @sfc-gh-kwagner . There are some tools to help do the static analysis, but what we have now is not enough for achieving that goal.

File: src/snowflake/connector/secret_detector.py
Comment (sfc-gh-hchaturvedi) [inline src/snowflake/connector/secret_detector.py:84]: <facepalm>. Good catch.

**Review by sfc-gh-hchaturvedi — APPROVED** [review]
LGTM!

**Pull request merged**
